### PR TITLE
Detect self-closing xml tags with attributes

### LINKED ISSUE #1166: XDocument BeEquivalentTo assertion returns incorrect path to node
### Description

XDocument BeEquivalentTo assertion returns incorrect path to node

### Complete minimal example reproducing the issue

```csharp
// Arrange
XDocument expected = XDocument.Parse("<?xml version=\"1.0\" encoding=\"utf-8\"?><root><a><aa><aaa attr=\"x\"/></aa></a><b><bb id=\"\"/></b></root>");
XDocument actual = XDocument.Parse("<?xml version=\"1.0\" encoding=\"utf-8\"?><root><a><aa><aaa attr=\"x\"/></aa></a><b><bb id=\"value\"/></b></root>");

// Assert
actual.Should().BeEquivalentTo(expected);
```

### Expected behavior:

Assertion should fail with the following message:
>Expected attribute "id" at "/root/b/bb" to have value "", but found "value".

### Actual behavior:

Assertion fails with the following message:
>Expected attribute "id" at "/root/a/b/bb" to have value "", but found "value".

The 'a' node is not in the correct path to the difference between the two xmls.

### Versions

* Version of Fluent Assertions: 5.9.0
* .NET framework 4.7

### Additional Information

If the attr attributes are removed, that is, the xml is modified to the following format:
```csharp
XDocument expected = XDocument.Parse("<?xml version=\"1.0\" encoding=\"utf-8\"?><root><a><aa><aaa/></aa></a><b><bb id=\"\"/></b></root>");
XDocument actual = XDocument.Parse("<?xml version=\"1.0\" encoding=\"utf-8\"?><root><a><aa><aaa/></aa></a><b><bb id=\"value\"/></b></root>");
```
then the test behaves as expected, the message is: 
>Expected attribute "id" at "/root/b/bb" to have value "", but found "value".

### PR DESCRIPTION
**Pull request by jnyrup**
When traversing the structure of an `XDocument` and into
the `Attributes` of an xml tag, we need to move out again
using `MoveToElement()` to properly detect the xml tag,
and not the attributes, as self-closing.
This fixes #1166

### CODE REVIEW

File: Tests/Shared.Specs/XDocumentAssertionSpecs.cs [hunk 1]
```diff
@@ -659,6 +659,21 @@ public void When_asserting_an_xml_document_is_equivalent_to_different_xml_docume
...
         }
 
+        [Fact]
+        public void when_an_xml_element_has_attributes_and_is_self_closing_it_should_succeed()
+        {
+            // Arrange
+            XDocument actual = XDocument.Parse("<xml><a attr=\"x\"/><b id=\"foo\"/></xml>");
+            XDocument expected = XDocument.Parse("<xml><a attr=\"x\"/><b id=\"bar\"/></xml>");
+
+            // Act
```
Comment (dennisdoomen) [inline Tests/Shared.Specs/XDocumentAssertionSpecs.cs]: Maybe we should start using this "light" AAA format everywhere

Comment (jnyrup) [reply Tests/Shared.Specs/XDocumentAssertionSpecs.cs]: Pure muscle memory, cause the the light AAA format is what I use at work.
I'll push an update to this PR to use the other format for consistency.

But I'll be happy to create a another PR converting to light AAA syntax. I don't recall having seen this heavier syntax anywhere else.

Comment (dennisdoomen) [reply Tests/Shared.Specs/XDocumentAssertionSpecs.cs]: I introduced the heavy format myself years ago, but I don't see a lot of value in it anymore.

Comment (danielmpetrov) [reply Tests/Shared.Specs/XDocumentAssertionSpecs.cs]: +1 on the 'light' AAA syntax. I find the current format over the top, and many times the comment to code ratio is very high, making it extra hard to navigate around the tests.

**Pull request merged**